goon_test.go

```diff
@@ -892,24 +948,64 @@ func TestMigration(t *testing.T) {
 	}
 
 	// Run migration tests with both IgnoreFieldMismatch on & off
-	for i := 0; i < 2; i++ {
-		IgnoreFieldMismatch = (i == 0)
+	testcase := []struct {
+		name                string
+		ignoreFieldMismatch bool
+		src, dst            MigrationEntity
+	}{
```

Comment:
We should still test with `IgnoreFieldMismatch` on & off with all the variants, not just _NormalCache -> NormalCache_. Writing all of these out can get too verbose (8 cases), so we could still do a loop `for i := 0; i < 2; i++ {` like before and inside the loop add the 4 different src/dst combo types, with `IgnoreFieldMismatch` set based on the loop variable.

Reply:
Fixed dd18dfdcf507a3c9df03d209d940e71b82833e10...1c5058d623f2f06c29ab918d5b2e79bf6c916f53